mypy/meet.py

```diff
@@ -682,6 +682,11 @@ def visit_erased_type(self, t: ErasedType) -> ProperType:
     def visit_type_var(self, t: TypeVarType) -> ProperType:
         if isinstance(self.s, TypeVarType) and self.s.id == t.id:
             return self.s
+        # this may be alright on things other than just instances
+        elif isinstance(self.s, (Instance, NoneType, TupleType)):
+            return t.copy_modified(upper_bound=meet_types(t.upper_bound, self.s))
```

Comment:
Note to self: there's probably a "is concrete type" utility somewhere in mypy. Maybe in checking for subtypes of `type`.